### DISCUSSION TITLE
Get notified when a sensors' tick is skipped because of an already used run key

### DISCUSSION THREAD

**Discussion by devlouisrdc** [dagster-io/dagster#19231] [Q&A] [answered]
Hello,

We are using Dagster 1.4.5 mainly for data flows orchestration and file processing. For one of our applications, we are processing a file which is loaded by users in a specific folder. In order to detect when a file is stuck in the loading folder, we would like to be notified when a sensors' tick is skipped because the run key has already been used.

We thought about logging the run key in the sensors' logs thanks to the "Experimental schedule/sensor logging view" but as we deployed our environment on AWS ECS with three separated tasks for Dagster, Dagit and Daemon, it appears that this solution can't work in our case.

We are now wondering if it would be possible to retrieve the list of pasts run keys inside the sensor in order to compare the current run key to the history. This way we would be able to identify if the run key has already been used and, then, return an explicit skip reason. Unfortunately, we didn't find any way to do it.

Could someone help us?

**Comment by jamiedemaria** [comment, marked as answer]
Hey @devlouisrdc  the functionality you're looking for doesn't exist, but you can open a feature request for it! https://github.com/dagster-io/dagster/issues/new?assignees=&labels=type%3A+feature-request&projects=&template=request_a_feature.yml

**Reply by devlouisrdc** [reply]
Hi @jamiedemaria, thank you for your answer! I think this kind of thing could be interesting for many people. I'll open a feature request.
